osu.Game/Screens/Edit/Screens/Setup/Components/LabelledComponents/LabelledSliderBar.cs

```diff
@@ -0,0 +1,251 @@
+    {
+        private readonly Box box;
+        private readonly OsuSpriteText label;
+        private readonly OsuSpriteText bottomText;
+        private readonly SetupTickSliderBar sliderBar;
+
+        private const float label_container_width = 150;
+        private const float corner_radius = 15;
+        private const float default_label_text_size = 16;
+        private const float default_bottom_label_text_size = 12;
+        private const float normal_height = 60;
+        private const float default_label_padding = 15;
+        private const float default_top_padding = 10;
+        private const float default_bottom_padding = 12;
+        private const float default_left_slider_bar_padding = 20;
+        private const float default_right_slider_bar_padding = 35;
+
+        public event Action<float> SliderBarValueChanged;
+
+        public void TriggerSliderBarValueChanged(float newValue)
+        {
+            SliderBarValueChanged?.Invoke(newValue);
+        }
+
+        public float CurrentValue
+        {
+            get => sliderBar.Current.Value;
+            set => sliderBar.Current.Value = value;
+        }
+
+        private float sliderMinValue;
+        public float SliderMinValue
+        {
+            get => sliderMinValue;
+            set
+            {
+                sliderMinValue = value;
+                sliderBar.MinValue = value;
+            }
+        }
+
+        private float sliderMaxValue = 10;
+        public float SliderMaxValue
+        {
+            get => sliderMaxValue;
+            set
+            {
+                sliderMaxValue = value;
+                sliderBar.MaxValue = value;
+            }
+        }
+
+        private float sliderNormalPrecision = 1;
+        public float SliderNormalPrecision
+        {
+            get => sliderNormalPrecision;
+            set
+            {
+                sliderNormalPrecision = value;
+                sliderBar.NormalPrecision = value;
+            }
+        }
+
+        private float sliderAlternatePrecision = 1;
+        public float SliderAlternatePrecision
+        {
+            get => sliderAlternatePrecision;
+            set
+            {
+                sliderAlternatePrecision = value;
+                sliderBar.AlternatePrecision = value;
+            }
+        }
+
+        private string tooltipTextSuffix = "";
+        public string TooltipTextSuffix
+        {
+            get => tooltipTextSuffix;
+            set
+            {
+                tooltipTextSuffix = value;
+                sliderBar.TooltipTextSuffix = value;
+            }
+        }
+
+        public string LabelText
+        {
+            get => label.Text;
+            set => label.Text = value;
+        }
+
+        public string BottomLabelText
+        {
+            get => bottomText.Text;
+            set
+            {
+                bottomText.Text = value;
+                Height = normal_height + (value != "" ? 20 : 0);
```

Comment:
I don't like this. Make this component autosize on the Y-axis to do this automatically.